### ISSUE TITLE
training loss decreasing, training cross-entropy increasing?

### ISSUE THREAD

**Issue by MFajcik** [mosaicml/llm-foundry#306]
Hi there, 
I am investigating very strange issue... I am fine-tuning MPT 7b for 2,000 steps, the training loss is decreasing, but training cross entropy (and validation cross entropy) do keep increase!

I have also investigated samples from the model and they seem okay! What might be the issue???

# Training loss
![loss_train](https://github.com/mosaicml/llm-foundry/assets/31218150/72e18f16-702d-4be2-94a7-00664e0e2192)

# Training cross-entropy
![ce_train](https://github.com/mosaicml/llm-foundry/assets/31218150/27acb3a1-8d79-46ac-9c99-dee6c292fc6c)

Here is full training log
https://drive.google.com/file/d/1lt-XZKrOLBqmbQ2AD2lsMz9fBfYIMUhx/view?usp=drive_link

**Comment by MFajcik** [comment]
This is the validation "LanguageCrossEntropy". Same trend.

['12.8596', '13.3547', '14.2374', '15.6595', '17.0223', '16.7493', '18.1268', '19.3765', '20.7455', '21.6569']

**Comment by vchiley** [comment]
Your google drive link is private.

I just want to verify that you're running training on the main branch. We use to see a diff between train loss and measured CE, but it should have been fixed [here](https://github.com/mosaicml/llm-foundry/pull/288).

Are you using the `train_small` ds shown in the quick start guide? (this would explain the jagged loss curve; looks like epoch boundaries)
Note that is a toy quick start example. You're not meant to train on it for a long time.

**Comment by MFajcik** [comment]
Hi @vchiley, thanks for your response! 

No, I have my own dataset (for multi-source summarization) and wrote my own config. The nucleus sampling samples (from 2k steps model) look fine.  The jagged curve is likely caused by epochs, as the finetuning dataset is rather small (BS64, ~15k samples).

I cloned the branch somewhere around ~15 May. I will try to update the repo on Monday, and let's see where this gets us.
